nibabel/nicom/dicomwrappers.py

```diff
@@ -522,6 +546,9 @@ def image_shape(self):
         if stackid_tag in dim_seq:
             stackid_dim_idx = dim_seq.index(stackid_tag)
             frame_indices = np.delete(frame_indices, stackid_dim_idx, axis=1)
+        if has_derived:
+            # derived volume is included
+            frame_indices = np.delete(frame_indices, 1, axis=1)
```

Comment:
It is always the second entry?

Reply:
Based off my limited testing yes - however we should definitely keep an eye on this

Reply:
Can we use either a derived index or make an assertion that will fail loudly when the assumption is violated?